src/HttpAdapterDiscovery.php

```diff
@@ -32,38 +41,14 @@ class HttpAdapterDiscovery
     protected static $cache;
 
     /**
-     * Register an HTTP Adapter
-     *
-     * @param string $name
-     * @param string $class
-     */
-    public static function register($name, $class)
-    {
-        static::$cache = null;
-
-        static::$adapters[$name] = $class;
-    }
-
-    /**
      * Finds an HTTP Adapter
      *
-     * @return object
+     * @return HttpAdapter
      *
      * @throws NotFoundException
      */
     public static function find()
     {
-        // We have a cache
-        if (isset(static::$cache)) {
-            return static::$cache;
-        }
-
-        foreach (static::$adapters as $name => $class) {
-            if (class_exists($class)) {
-                return static::$cache = new $class;
-            }
-        }
-
-        throw new NotFoundException('No HTTP Adapter found');
+        return parent::find();
```

Comment:
Necessary, no. But (as you may have noticed elsewhere) I value IDE auto-completion very much.